test/avocado/selenium-login.py

```diff
@@ -189,5 +190,161 @@ def test30ChangeTabServices(self):
+        elem.click()
+        elem = self.wait_id('journal-entry')
+        self.mainframe()
+
+    def test60ChangeTabStorage(self):
+        reald_name = process.run(
+            "storagedctl status | tail -1 |sed -r 's/.* ([a-z]+).*/\\1/'", shell=True).stdout[:-1]
+        reald_serial = process.run(
+            "storagedctl status | tail -1 |sed -r 's/.* ([^ ]+)\s+[a-z]+.*/\\1/'", shell=True).stdout[:-1]
+        print ">>>" + reald_name + ">>>" + reald_serial + ">>>"
+        other_disc = libdisc.DiscSimple()
+        other_discname = other_disc.adddisc("d1")
+        other_shortname = os.path.basename(other_discname)
+        self.login()
+        self.wait_link('Storage').click()
+        self.wait_frame("storage")
+        self.wait_id("drives")
+        elem = self.wait_xpath("//*[@data-goto-block='%s']" % other_shortname, cond=clickable)
+        elem.click()
+        self.wait_id('storage-detail')
+        self.wait_xpath("//td[contains(text(), '%s')]" % other_discname)
+        self.wait_xpath("//td[contains(text(), '%s')]" % "Capacity")
+        self.wait_xpath("//td[contains(text(), '%s')]" % "1000 MB")
+
+        self.wait_link('Storage').click()
+        elem = self.wait_xpath("//*[@data-goto-block='%s']" % other_shortname)
+
+        self.mainframe()
+
+    def test70ChangeTabNetworking(self):
+        self.login()
+        out = process.run(
+            "ip r |grep default | head -1 | cut -d ' ' -f 5", shell=True)
+        self.wait_link('Network').click()
+        self.wait_frame("network")
+
+        self.wait_id("networking-interfaces")
+        self.wait_id("networking-tx-graph")
+
+        elem = self.wait_xpath("//tr[@data-interface='%s']" % out.stdout[:-1],cond=clickable)
+        elem.click()
+        self.wait_xpath("//td[contains(text(), '%s')]" % "Carrier")
+        self.mainframe()
+
+    def test80ChangeTabTools(self):
+        self.login()
+        self.wait_id("tools-panel",cond=invisible)
+        self.wait_link('Tools').click()
+        self.wait_id("tools-panel")
+        self.wait_link('Accounts').click()
+        self.wait_frame("users")
+        elem = self.wait_xpath(
+            "//*[@class='cockpit-account-user-name' and contains(text(), '%s')]" % user, cond=clickable)
+        elem.click()
+        elem = self.wait_id('account')
+        self.wait_xpath("//*[contains(text(), '%s')]" % "Full Name")
+        self.wait_link('Accounts', elem).click()
+        self.wait_xpath('//*[@id="accounts-create"]', cond=clickable).click()
+        elem = self.wait_id('accounts-create-real-name')
+        elem.clear()
+        elem.send_keys('testxx')
+        elem = self.wait_id('accounts-create-pw1')
+        elem.clear()
+        elem.send_keys(passwd)
+        elem = self.wait_id('accounts-create-pw2')
+        elem.clear()
+        elem.send_keys(passwd)
+        time.sleep(self.default_sleep)
+        self.wait_id('accounts-create-create', cond=clickable).click()
+        time.sleep(self.default_sleep)
```

Comment:
We shouldn't be using sleeps here. This will make the tests very unstable. We need to just wait until the UI updates for the given condition. If the UI does not give enough indication, then the UI should be modified to set an attribute or only display the text when ready.